ssh.go

```diff
@@ -43,16 +44,27 @@ func (e ErrConnect) Error() string {
 
 // parseHost parses and normalizes <user>@<host:port> from a given string.
 func (c *SSHClient) parseHost(host string) error {
-	c.host = host
+	// https://golang.org/pkg/net/url/#URL
+	// [scheme:][//[userinfo@]host][/]path[?query][#fragment]
+	if !strings.Contains(host, "://") && !strings.HasPrefix(host, "//") {
+		host = "ssh://" + host
+	}
+
+	hostURL, err := url.Parse(host)
```

Comment:
Can you test this with:
```
127.0.0.1
localhost
localhost:22
ssh://localhost
```
etc.?

I know `url.Parse(host)` had some issues with localhost URLs.

Reply:
@VojtechVitek  what is the  issues?

here is a simple test:
```
sup kadefor$ cat ssh_test.go
package sup

import (
	"fmt"
	"testing"
)

func Test_parseHost(t *testing.T) {
	hosts := []string{
		"127.0.0.1",
		"localhost",
		"localhost:22",
		"ssh://localhost",

		"tom@127.0.0.1",
		"tom@localhost",
		"tom@localhost:22",
		"ssh://tom@localhost",

		"tom:11@@33@127.0.0.1",
		"tom:11@@33@localhost",
		"tom:11@@33@localhost:22",
		"ssh://tom:11@@33@localhost",
	}

	for _, host := range hosts {
		c := &SSHClient{}
		err := c.parseHost(host)
		if err != nil {
			fmt.Println(err)
		} else {
			fmt.Printf("%s\n\t%v\n", host, c)
		}
	}
}
```

```
sup kadefor$ go test
127.0.0.1
	&{<nil> <nil> kadefor 127.0.0.1:22 <nil> <nil> <nil> false false false export SUP_HOST="127.0.0.1:22"; }
localhost
	&{<nil> <nil> kadefor localhost:22 <nil> <nil> <nil> false false false export SUP_HOST="localhost:22"; }
localhost:22
	&{<nil> <nil> kadefor localhost:22 <nil> <nil> <nil> false false false export SUP_HOST="localhost:22"; }
ssh://localhost
	&{<nil> <nil> kadefor localhost:22 <nil> <nil> <nil> false false false export SUP_HOST="localhost:22"; }
tom@127.0.0.1
	&{<nil> <nil> tom 127.0.0.1:22 <nil> <nil> <nil> false false false export SUP_HOST="127.0.0.1:22"; }
tom@localhost
	&{<nil> <nil> tom localhost:22 <nil> <nil> <nil> false false false export SUP_HOST="localhost:22"; }
tom@localhost:22
	&{<nil> <nil> tom localhost:22 <nil> <nil> <nil> false false false export SUP_HOST="localhost:22"; }
ssh://tom@localhost
	&{<nil> <nil> tom localhost:22 <nil> <nil> <nil> false false false export SUP_HOST="localhost:22"; }
tom:11@@33@127.0.0.1
	&{<nil> <nil> tom 127.0.0.1:22 <nil> <nil> <nil> false false false export SUP_HOST="127.0.0.1:22"; }
tom:11@@33@localhost
	&{<nil> <nil> tom localhost:22 <nil> <nil> <nil> false false false export SUP_HOST="localhost:22"; }
tom:11@@33@localhost:22
	&{<nil> <nil> tom localhost:22 <nil> <nil> <nil> false false false export SUP_HOST="localhost:22"; }
ssh://tom:11@@33@localhost
	&{<nil> <nil> tom localhost:22 <nil> <nil> <nil> false false false export SUP_HOST="localhost:22"; }
PASS
ok  	github.com/pressly/sup	0.014s
```